### ISSUE TITLE
Hotswap agent locks files even after undeploy causing mvn clean to fail

### ISSUE THREAD

**Issue by eitland** [HotswapProjects/HotswapAgent#144]
When I run Tomee server using dcevm hand hotswap agent it seems to somehow keep a lock on some of the files even after I undeploy the application from the server.

This causes mvn clean to fail.

I have not been able to recreate it with only DCEVM (i.e. without hotswap agent enabled).

**Comment by skybber** [comment]
Should be solved by  #127

**Comment by eitland** [comment]
Thanks! Cloning to test and verify now.

**Comment by eitland** [comment]
Short update:

I have tested on my current project. Preliminary comclusion:  It works:

What I have tested (once so far):

- [x] 1 Undeploy project, clean-and-build. Was broken for me, now works.
- [x] 2 Redeploy said project
- [x] 3 Hotswapping a method works as before

For now I get lots of error messages along the lines of 

`HOTSWAP AGENT: 13:14:42.689 ERROR (org.hotswap.agent.watch.nio.EventDispatcher) - No match for  watch event 'sun.nio.fs.AbstractWatchKey$Event@78405fb6',  path 'c:\...'`

I haven't thought to much about it but expect that it just means I have to adjust logging settings somewhere.
